src/index.ts

```diff
@@ -58,19 +72,22 @@ export function getDataSuffix({
   }
 
   // Encode the data according to ABI encoding rules
+  const encodedUser = encodeAddress(user)
   const encodedConsumer = encodeAddress(consumer)
   const encodedProviders = encodeAddressArray(providers)
-  const encodedBytes = encodedConsumer + encodedProviders
-
-  // Calculate the total length of the data (in bytes)
-  const totalLength = (8 + 2 + encodedBytes.length + 8) / 2 // 8 for prefix, 2 for format byte, rest for data and length
-  const lengthHex = totalLength.toString(16).padStart(8, '0')
+  const encodedBytes = encodedUser + encodedConsumer + encodedProviders
 
-  // Get the format byte
-  const formatByte = FORMAT_ID_BYTES['default']
+  // Calculate the length of the payload (in bytes)
+  const payloadLength = encodedBytes.length / 2
+  const payloadLengthHex = payloadLength.toString(16).padStart(4, '0')
```

Comment:
i don't fully understand the current logic and this change makes sense, but wanted to check that it's intentionally different? we no longer seem to account for the prefix and format bytes here